src/codecs/hdr/decoder.rs

```diff
@@ -460,21 +459,16 @@ impl<R: BufRead> HdrDecoder<R> {
         }
 
         let chunks_iter = output_slice.chunks_mut(self.width as usize);
-        let mut pool = Pool::new(8); //
-
-        (pool.scoped(|scope| {
-            for chunk in chunks_iter {
-                let mut buf = vec![Default::default(); self.width as usize];
-                read_scanline(&mut self.r, &mut buf[..])?;
-                let f = &f;
-                scope.execute(move || {
-                    for (dst, &pix) in chunk.iter_mut().zip(buf.iter()) {
-                        *dst = f(pix);
-                    }
-                });
+
+        let mut buf = vec![Default::default(); self.width as usize];
+        for chunk in chunks_iter {
+            read_scanline(&mut self.r, &mut buf[..])?;
+            for (dst, &pix) in chunk.iter_mut().zip(buf.iter()) {
+                *dst = f(pix);
             }
-            Ok(())
-        }) as Result<(), ImageError>)?;
+            buf.clear();
+            buf.resize(self.width as usize, Default::default());
```

Comment:
Would `buf.fill(Default::default())` work instead?

Reply:
Actually, why is this needed at all? Shouldn't read_scanline overwrite the entire buffer?

Reply:
Removed (and commented) in [7df5cb1](https://github.com/image-rs/image/pull/1857/commits/7df5cb142318525ff1c544361656157c6bb70be6)